nilearn/connectome/connectivity_matrices.py

```diff
@@ -533,9 +537,15 @@ def _fit_transform(
         if self.cov_estimator is None:
             self.cov_estimator = LedoitWolf(store_precision=False)
 
+        if not hasattr(X, "__iter__"):
+            raise TypeError(
+                "Input must be an iterable of numpy arrays. "
+                f"Got {X.__class__.__name__}"
+            )
+
         # casting to a list
         # to make it easier to check with sklearn estimator compliance
-        if not isinstance(X, list):
+        if isinstance(X, np.ndarray) and X.ndim == 2:
```

Comment:
I guess  `_fit_transform` is only called from `fit_transform`. In that case, at this point we should be sure that X is of expected type as these checks are already done in `fit_transform`.

Reply:
Ok, sorry, I did not pay attention to `do_fit` and `do_transform` parameters.